fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -707,28 +713,44 @@ public static Type invertComparisonType(@Nonnull final Comparisons.Type type) {
     @Nullable
     @SpotBugsSuppressWarnings("NP_BOOLEAN_RETURN_NULL")
     public static Boolean evalComparison(@Nonnull Type type, @Nullable Object value, @Nullable Object comparand) {
-        if (value == null) {
-            return null;
-        }
```

Comment:
as I mentioned in another comment, because of this is called first, we'll never reach the `if(fieldValue == null)` block in method `compare`. I removed this block here, and added to cases below.

Reply:
right now, if type = TEXT_CONTAINS_ALL, or anything that is not in the `switch` block, and `value == null`, this method returns null. Is this the expected behavior?

Reply:
👍 after removing the `if (value == null) return null` block, unknown type throws exception regardless of input value being null.